MyObjects/Helpers/MyWebDriver.cs

```diff
@@ -0,0 +1,117 @@
+﻿using NUnit.Framework;
+using NUnit.Framework.Interfaces;
+using OpenQA.Selenium;
+using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace MyObjects.Helpers
+{
+    public class MyWebDriver : IWebDriver
+    {
+        public IWebDriver webDriver;
+
+        public MyWebDriver(IWebDriver webDriver)
+        {
+            this.webDriver = webDriver;
+        }
+
+        public IWebElement FindElement(By locator)
+        {
+            Logger.Info($"Searching for element by locator: {locator}");
+            return webDriver.FindElement(locator);
+        }
+
+        public ReadOnlyCollection<IWebElement> FindElements(By locator)
+        {
+            Logger.Info($"Searching for element by locator: {locator}");
+            return webDriver.FindElements(locator);
```

Comment:
you can also do webDriver.FindElements(locator).ToList() and change return type. this will make further usages easier

Reply:
in that case, IWebDriver interface method will not be implemented
![image](https://user-images.githubusercontent.com/112108550/204535906-340070c3-8832-4f1f-b1ad-414969891b20.png)